x-pack/solutions/search/plugins/search_connectors/server/task.ts

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger, CoreStart, SavedObjectsClient } from '@kbn/core/server';
+
+import type {
+  ConcreteTaskInstance,
+  TaskManagerStartContract,
+  TaskInstance,
+} from '@kbn/task-manager-plugin/server';
+
+import { LicensingPluginStart } from '@kbn/licensing-plugin/server';
+import type {
+  SearchConnectorsPluginStartDependencies,
+  SearchConnectorsPluginSetupDependencies,
+} from './types';
+import {
+  AgentlessConnectorsInfraService,
+  getConnectorsWithoutPolicies,
+  getPoliciesWithoutConnectors,
+} from './services';
+
+import { SearchConnectorsConfig } from './config';
+
+const AGENTLESS_CONNECTOR_DEPLOYMENTS_SYNC_TASK_ID = 'search:agentless-connectors-sync-task';
+const AGENTLESS_CONNECTOR_DEPLOYMENTS_SYNC_TASK_TYPE = 'search:agentless-connectors-sync';
+
+const SCHEDULE = { interval: '1m' };
+
+export function infraSyncTaskRunner(
```

Comment:
potential followup: It would be cool if we could force schedule this when e.g. a user creates a new connector. This would limit the wait time for the infrastructure to get deployed.

Reply:
Yes! Should be easy, because I already actually schedule this task during plugin startup - doing it again is easy and will require really minor refactoring.